pkg/chunked/storage_linux.go

```diff
@@ -1141,6 +1141,32 @@ func makeEntriesFlat(mergedEntries []fileMetadata) ([]fileMetadata, error) {
 	return new, nil
 }
 
+// checkNoStreams checks that there are no other streams on the channel and that there are no errors.
+func checkNoStreams(streams chan io.ReadCloser, errs chan error) error {
+	var err error
+	closed := 0
+	for {
+		select {
+		case _, ok := <-streams:
+			if !ok {
+				closed++
```

Comment:
AFAICS once this happens, this `case` will _always_ trigger and this function will consume 100% of a CPU until the other channel is closed.

Reply:
… and `close` will keep incrementing.